src/main/kotlin/com/atlassian/performance/tools/report/api/distribution/DistributionComparator.kt

```diff
@@ -0,0 +1,113 @@
+     */
+    private val tolerance: Double,
+    private val significance: Double
+) {
+
+
+
+    /**
+     * Performs a one-tailed Mann–Whitney U test to check whether experiment is not slower than the baseline
+     *
+     * @return true if the experiment is slower than the baseline by more than tolerance, false otherwise
+     */
+    private fun isExperimentRegressed(baselineMedian: Double): Boolean {
+        val mu = -tolerance * baselineMedian
+        return WilcoxonRankSum(baseline, experiment, mu).pValue1SidedLess < significance
+    }
+
+    private fun isExperimentImproved(baselineMedian: Double): Boolean {
+        val mu = -tolerance * baselineMedian
+        val wilcoxon = WilcoxonRankSum(experiment, baseline, mu)
+        return wilcoxon.pValue1SidedLess < significance
+    }
+
+    /**
+     * Pseudo-median: the median of the Walsh (pairwise) averages
+     */
+    private fun pseudoMedian(array: DoubleArray): Double {
+        val n = array.size
+        val size = n * (n + 1) / 2 - n
+        val values = DoubleArray(size)
+        var k = 0
+        for (i in 0 until n) {
+            for (j in i + 1 until n) {
+                values[k++] = (array[i] + array[j]) / 2
+            }
+        }
+        return Median().evaluate(values)
+    }
+
+    private fun median(func: (xi: Double, yj: Double) -> Double): Double {
+        val values = DoubleArray(baseline.size * experiment.size)
+        var k = 0
+        for (i in baseline.indices) {
+            for (j in experiment.indices) {
+                values[k++] = func(baseline[i], experiment[j])
+            }
+        }
+        return Median().withNaNStrategy(NaNStrategy.MINIMAL).evaluate(values)
+    }
+
+    private fun shift(): Double {
+        return median { xi, yj -> yj - xi }
+    }
+
+    private fun ratio(): Double {
+        return median { xi, yj -> yj / xi }
+    }
+
+    /**
+     * Calculates the distance between two data sets based on the [Hodges-Lehmann estimator][].
+     * [Hodges-Lehmann estimator]: https://en.wikipedia.org/wiki/Hodges%E2%80%93Lehmann_estimator
+     * https://aakinshin.net/hodges-lehmann-estimator/
+     * https://github.com/AndreyAkinshin/perfolizer/blob/master/src/Perfolizer/Perfolizer/Mathematics/GenericEstimators/HodgesLehmannEstimator.cs
+     *
+     * Takes into account tolerance which answers the question "is change is big enough to matter?"
+     */
+    fun compare(): DistributionComparison {
+        val experimentShift = shift()
+        val baselineMedian = pseudoMedian(baseline)
+        val experimentRatio = ratio()
+        val isExperimentImproved = isExperimentImproved(baselineMedian)
+        val isExperimentRegressed = isExperimentRegressed(baselineMedian)
+        val experimentRelativeChange = experimentRatio - 1
+        return DistributionComparison(
+            experimentRelativeChange = experimentRelativeChange,
+            experimentAbsoluteChange = experimentShift,
```

Comment:
No need to introduce two new terms for the same thing. Let's use https://martinfowler.com/bliki/UbiquitousLanguage.html

```suggestion
            experimentShift = experimentShift,
```
And inline the value.